wicket-core/src/test/java/org/apache/wicket/protocol/http/request/WebClientInfoTest.java

```diff
@@ -59,9 +58,52 @@ public void before()
 	}
 	
 	/**
-	 * Test IE 6.x user-agent strings
+	 * Test check check all user agents
 	 */
 	@Test
+	public void testBrowsersAndVersionsOfUserAgents() {
+		// Internet Explorers
```

Comment:
I would rather have all these as individual tests, to be able to see which one was failed
And run all of them regardless failure state of others

Reply:
This might slow down the test execution because of the init of YAUAA - try it out.

Reply:
You will see if something fails in the test results btw.

Reply:
You can move init to `@BeforeClass` .....

Reply:
Yes that is true, but I think if you change the code / the browser detection you are going to investigate why the test is failing and fix the integration as long as every test is back working again.

@BeforeClass is not working, because WicketTestCase and at this stage no Application is attached to the main thread.

Reply:
May you have any other good idea to keep the tests individual.

Just send a patch to me if something would fit the requirement and I am going to integrate it.

Reply:
Will try to provide patch later tonight
Have to do day time job right now :(

Reply:
Hehe me too 😄 - no hurry, because I think this is going to be released in a further version.

Reply:
Sorry for delay, here is the code works for me:

```
	private static ThreadLocal<UserAgentAnalyzer> localAnalyzer = new ThreadLocal<UserAgentAnalyzer>();

	@Before
	public void before()
	{
		requestCycleMock = mock(RequestCycle.class);

		webRequest = mock(ServletWebRequest.class);
		when(requestCycleMock.getRequest()).thenReturn(webRequest);

		servletRequest = mock(HttpServletRequest.class);
		when(webRequest.getContainerRequest()).thenReturn(servletRequest);

		if (localAnalyzer.get() == null) {
			WebClientInfo webClientInfo = new WebClientInfo(requestCycleMock, "test");
			webClientInfo.gatherExtendedInfo();
			localAnalyzer.set(Application.get().getMetaData(WebClientInfo.UAA_META_DATA_KEY));
		} else {
			Application.get().setMetaData(WebClientInfo.UAA_META_DATA_KEY, localAnalyzer.get());
		}
	}
	
	@AfterClass
	public static void staticAfter()
	{
		localAnalyzer.set(null);
	}
```

Here are the measurment results:
```
Combined tests
real	0m24.123s
user	1m30.748s
sys	0m1.804s


Individual tests (after above changes were applied)
real	0m22.844s
user	1m23.948s
sys	0m1.832s

```